Installation issue: axom

### Steps to reproduce the issue

```console
spack install axom@0.5.0
```

### Information on your system

* **Spack:** 0.16.2-4207-5fddd48f80
* **Python:** 3.7.2
* **Platform:** linux-rhel7-broadwell
* **Concretizer:** original

### Additional information

axom@0.5.0 needs umpire@5.0.0 but package is not constrained and fails using umpire@6.0.0

@white238 @davidbeckingsale.

Relevant part of the log:
```
     580     make[2]: Entering directory '/usr/WS2/bernede1/gitlab-runner/build
             s/llnl-stack-19d51ded/spack/var/spack/stage/spack-stage-axom-0.5.0
             -bnpgq3mmbjcedrul2k7xtoyturx47cht/spack-build-bnpgq3m'
     581     In file included from /usr/WS2/bernede1/gitlab-runner/builds/llnl-
             stack-19d51ded/spack/var/spack/stage/spack-stage-axom-0.5.0-bnpgq3
             mmbjcedrul2k7xtoyturx47cht/spack-src/src/axom/quest/MeshTester.cpp
             :7:
  >> 582     /usr/WS2/bernede1/gitlab-runner/builds/llnl-stack-19d51ded/spack/v
             ar/spack/stage/spack-stage-axom-0.5.0-bnpgq3mmbjcedrul2k7xtoyturx4
             7cht/spack-src/src/axom/quest/MeshTester.hpp:128:40: error: no mem
             ber named 'DynamicPool' in namespace 'umpire::strategy'
     583         rm.makeAllocator<umpire::strategy::DynamicPool>(
     584                          ~~~~~~~~~~~~~~~~~~^
  >> 585     /usr/WS2/bernede1/gitlab-runner/builds/llnl-stack-19d51ded/spack/v
             ar/spack/stage/spack-stage-axom-0.5.0-bnpgq3mmbjcedrul2k7xtoyturx4
             7cht/spack-src/src/axom/quest/MeshTester.hpp:128:52: error: expect
             ed unqualified-id
     586         rm.makeAllocator<umpire::strategy::DynamicPool>(
     587                                                        ^
```

[spack-build-out (2).txt](https://github.com/spack/spack/files/7138906/spack-build-out.2.txt)


### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/axom/package.py
+++ b/var/spack/repos/builtin/packages/axom/package.py
@@ -107,14 +107,15 @@ class Axom(CachedCMakePackage, CudaPackage):
     depends_on("raja+openmp", when="+raja+openmp")
     depends_on("raja+cuda", when="+raja+cuda")
 
-    depends_on("umpire@5.0.1:5~openmp", when="+umpire~openmp")
-    depends_on("umpire@5.0.1:5+openmp", when="+umpire+openmp")
-    depends_on("umpire@5.0.1:5+cuda", when="+umpire+cuda")
+    with when('+umpire'):
+        depends_on('umpire@5.0.1:5')
+        depends_on('umpire +openmp', when='+openmp')
+        depends_on('umpire +cuda', when='+cuda')
 
     for sm_ in CudaPackage.cuda_arch_values:
         depends_on('raja cuda_arch={0}'.format(sm_),
                    when='+raja cuda_arch={0}'.format(sm_))
-        depends_on('umpire@5.0.1:5 cuda_arch={0}'.format(sm_),
+        depends_on('umpire cuda_arch={0}'.format(sm_),
                    when='+umpire cuda_arch={0}'.format(sm_))
 
     depends_on("mfem", when="+mfem")

--- a/var/spack/repos/builtin/packages/axom/package.py
+++ b/var/spack/repos/builtin/packages/axom/package.py
@@ -107,14 +107,14 @@ class Axom(CachedCMakePackage, CudaPackage):
     depends_on("raja+openmp", when="+raja+openmp")
     depends_on("raja+cuda", when="+raja+cuda")
 
-    depends_on("umpire~openmp", when="+umpire~openmp")
-    depends_on("umpire+openmp", when="+umpire+openmp")
-    depends_on("umpire+cuda", when="+umpire+cuda")
+    depends_on("umpire@5.0.1~openmp", when="+umpire~openmp")
+    depends_on("umpire@5.0.1+openmp", when="+umpire+openmp")
+    depends_on("umpire@5.0.1+cuda", when="+umpire+cuda")
 
     for sm_ in CudaPackage.cuda_arch_values:
         depends_on('raja cuda_arch={0}'.format(sm_),
                    when='+raja cuda_arch={0}'.format(sm_))
-        depends_on('umpire cuda_arch={0}'.format(sm_),
+        depends_on('umpire@5.0.1 cuda_arch={0}'.format(sm_),
                    when='+umpire cuda_arch={0}'.format(sm_))
 
     depends_on("mfem", when="+mfem")

--- a/var/spack/repos/builtin/packages/axom/package.py
+++ b/var/spack/repos/builtin/packages/axom/package.py
@@ -107,14 +107,14 @@ class Axom(CachedCMakePackage, CudaPackage):
     depends_on("raja+openmp", when="+raja+openmp")
     depends_on("raja+cuda", when="+raja+cuda")
 
-    depends_on("umpire@5.0.1~openmp", when="+umpire~openmp")
-    depends_on("umpire@5.0.1+openmp", when="+umpire+openmp")
-    depends_on("umpire@5.0.1+cuda", when="+umpire+cuda")
+    depends_on("umpire@5.0.1:5~openmp", when="+umpire~openmp")
+    depends_on("umpire@5.0.1:5+openmp", when="+umpire+openmp")
+    depends_on("umpire@5.0.1:5+cuda", when="+umpire+cuda")
 
     for sm_ in CudaPackage.cuda_arch_values:
         depends_on('raja cuda_arch={0}'.format(sm_),
                    when='+raja cuda_arch={0}'.format(sm_))
-        depends_on('umpire@5.0.1 cuda_arch={0}'.format(sm_),
+        depends_on('umpire@5.0.1:5 cuda_arch={0}'.format(sm_),
                    when='+umpire cuda_arch={0}'.format(sm_))
 
     depends_on("mfem", when="+mfem")

